mithril-aggregator/src/database/provider/signer.rs

```diff
@@ -503,31 +660,90 @@ mod tests {
 
         for signer_record in signer_records_fake.clone() {
             store_recorder
-                .record_signer_id(signer_record.signer_id.clone())
+                .record_signer_registration(signer_record.signer_id.clone())
                 .await
-                .expect("record_signer_id should not fail");
+                .expect("record_signer_registration should not fail");
             let connection = &*connection.lock().await;
             let provider = SignerRecordProvider::new(connection);
             let signer_records_stored: Vec<SignerRecord> = provider
                 .get_by_signer_id(signer_record.signer_id)
                 .unwrap()
                 .collect::<Vec<_>>();
             assert_eq!(1, signer_records_stored.len());
+            assert!(
+                signer_records_stored
+                    .iter()
+                    .all(|s| s.registered_at.is_some()),
+                "registering a signer should set the registration date"
+            )
         }
+    }
+
+    #[tokio::test]
+    async fn test_store_import_signer() {
+        let signer_records_fake = fake_signer_records(5);
+
+        let connection = Connection::open(":memory:").unwrap();
+        setup_signer_db(&connection, Vec::new()).unwrap();
+
+        let connection = Arc::new(Mutex::new(connection));
+        let store = SignerStore::new(connection.clone());
 
         for signer_record in signer_records_fake {
-            let pool_ticker = Some(format!("new-pool-{}", signer_record.signer_id));
-            store_recorder
-                .record_signer_pool_ticker(signer_record.signer_id.clone(), pool_ticker.clone())
+            store
+                .import_signer(
+                    signer_record.signer_id.clone(),
+                    signer_record.pool_ticker.clone(),
+                )
                 .await
-                .expect("record_signer_pool_ticker should not fail");
+                .expect("import_signer should not fail");
             let connection = &*connection.lock().await;
             let provider = SignerRecordProvider::new(connection);
             let signer_records_stored: Vec<SignerRecord> = provider
                 .get_by_signer_id(signer_record.signer_id)
                 .unwrap()
                 .collect::<Vec<_>>();
-            assert_eq!(pool_ticker, signer_records_stored[0].to_owned().pool_ticker);
+            assert_eq!(
+                signer_record.pool_ticker,
+                signer_records_stored[0].to_owned().pool_ticker
+            );
+            assert!(
+                signer_records_stored
+                    .iter()
+                    .all(|s| s.registered_at.is_none()),
+                "imported signer should not have a registration date"
+            )
         }
     }
+
+    #[tokio::test]
+    async fn test_store_import_many_signers() {
+        let signers_fake: HashMap<_, _> = fake_signer_records(5)
```

Comment:
This will probably avoid flakiness with this test
```suggestion
        let signers_fake: BTreeMap<_, _> = fake_signer_records(5)
```